crates/ty_python_semantic/src/types.rs

```diff
@@ -4276,14 +4304,24 @@ impl<'db> Type<'db> {
                 .into()
             }
 
-            Type::GenericAlias(_) => {
+            Type::GenericAlias(alias) => {
+                let instantiated = Type::instance(db, ClassType::from(alias));
+
+                let parameters = if alias.origin(db).is_known(db, KnownClass::Tuple) {
```

Comment:
I think I follow the constructors here, but can you put a comment with the Python syntax of the signature you're trying to construct here?